### PR TITLE
Add basic internal type checks

### PR DESCRIPTION
**Pull request by jkrems**
I went for a minimal PR with internal checks only as a starting point. May follow up with type checks for the tests and/or public interface types.

### CODE REVIEW

File: lib/respond.js [hunk 1]
```diff
@@ -103,12 +144,13 @@ class VirtualResponse extends PassThrough {
     if (body === null) body = Buffer.alloc(0);
 
     if (body instanceof Buffer) {
+      // @ts-ignore
       this.body = body;
```
Comment (jkrems) [inline lib/respond.js]: I wish we could get rid of this but it's a breaking change I guess..? Being able to immediately read the body without piping is neat but I'm not sure it's worth keeping this super awkward "overwrite the helper method" around for.

File: lib/respond.js [hunk 2]
```diff
@@ -142,14 +188,24 @@ class VirtualResponse extends PassThrough {
     });
   }
 
+  /**
+   * @template {NodeJS.WritableStream} T
+   * @param {T} res
+   * @param {{ end?: boolean | undefined } | undefined} [options]
```
Comment (dbushong) [inline lib/respond.js]: does this need the second `| undefined` given that `options` is already optional?

File: package.json [hunk 3]
```diff
@@ -28,6 +28,9 @@
     "caseless": "^0.10.0"
   },
   "devDependencies": {
+    "@types/caseless": "^0.12.2",
+    "@types/mocha": "^5.2.7",
+    "@types/node": "^12.0.4",
     "assertive": "^2.1.0",
```
Comment (dbushong) [inline package.json]: looks like we also need a `npm i -D typescript`

**Comment by jkrems** [comment]
Fixed!

File: lib/respond.js (hunk 2 above)
Comment (jkrems) [reply lib/respond.js]: Removed both `| undefined` since they were redundant.

**Pull request merged**